django_loci/static/django-loci/js/loci.js

```diff
@@ -377,4 +497,120 @@ django.jQuery(function ($) {
         $indoor.show();
         $indoorRows.show();
     }
+
+    window.geocodeApi = window.location.origin + '/admin/django_loci/location/geocode/';
+    window.reverseApi = window.location.origin + '/admin/django_loci/location/reverse-geocode/';
+    window.setTimeout(function () {
+        window.addressLat = null;
+        window.addressLng = null;
+        window.receivedAlert = false;
+        window.skip_geocode = 0;  // amount of times geocode operation should be not executed
+        window.markerCoords = getMarker() !== undefined ? getMarker().getLatLng() : null;
+        $addressInput.change(function () {
+            if ($addressInput.val().length >= 10) {
+                updateMap(window.geocodeApi);
+            }
+        });
+        var timeout = null;
+        $addressInput.keyup(function () {
+            window.clearTimeout(timeout);
+            timeout = window.setTimeout(function () {
+                if ($addressInput.val().length >= 10) {
+                    updateMap(window.geocodeApi);
+                }
+            }, 1000);
+        });
+
+        window.setInterval(function () {
```

Comment:
can you explain what this code is doing?
It looks like you're periodically checking if the marker has coordinates, but I think we can listen to leaflet events instead of doing it this way.

I have tested this and doesn't really work as expected: it doesn't ask me for confirmation to update the address if the address field is already filled.
Also if I delete the marker I noticed a JS error that breaks this feature completely:
```
loci.js:443 Uncaught TypeError: Cannot read property 'dragging' of undefined
    at updateAdress (loci.js:443)
```

Reply:
> I have tested this and doesn't really work as expected: it doesn't ask me for confirmation to update the address if the address field is already filled.

It asks you after hitting save (django save), but I am not sure what do you mean. Could you give me step by step how to reproduce?

> Also if I delete the marker I noticed a JS error that breaks this feature completely:

same here

Reply:
not after django save.. after the widget of django-leaflet save, as explained in this comment: https://github.com/openwisp/django-loci/pull/46#pullrequestreview-347817033

Reply:
> Also if I delete the marker I noticed a JS error that breaks this feature completely:
loci.js:443 Uncaught TypeError: Cannot read property 'dragging' of undefined
    at updateAdress (loci.js:443)

I am not able to reproduce this part anyway, make sure that code is rebased